plugins/woocommerce/e2e/tests/shopper/my-account.spec.js

```diff
@@ -22,22 +22,28 @@ test.describe( 'My account page', () => {
 
 		// assert that navigation is visible
 		await expect(
-			page.locator( '.woocommerce-MyAccount-navigation-link >> nth=0' )
+			page.locator( '.woocommerce-MyAccount-navigation-link--dashboard' )
 		).toContainText( 'Dashboard' );
 		await expect(
-			page.locator( '.woocommerce-MyAccount-navigation-link >> nth=1' )
+			page.locator( '.woocommerce-MyAccount-navigation-link--orders' )
 		).toContainText( 'Orders' );
 		await expect(
-			page.locator( '.woocommerce-MyAccount-navigation-link >> nth=2' )
+			page.locator( '.woocommerce-MyAccount-navigation-link--downloads' )
 		).toContainText( 'Downloads' );
 		await expect(
-			page.locator( '.woocommerce-MyAccount-navigation-link >> nth=3' )
+			page.locator(
+				'.woocommerce-MyAccount-navigation-link--edit-address'
+			)
 		).toContainText( 'Addresses' );
 		await expect(
-			page.locator( '.woocommerce-MyAccount-navigation-link >> nth=4' )
+			page.locator(
+				'.woocommerce-MyAccount-navigation-link--edit-account'
+			)
 		).toContainText( 'Account details' );
 		await expect(
-			page.locator( '.woocommerce-MyAccount-navigation-link >> nth=5' )
+			page.locator(
+				'.woocommerce-MyAccount-navigation-link--customer-logout'
+			)
```

Comment:
Solid improvement.  I had structured it this way because it takes into account the order of the links displayed -- but that isn't really important.  I always prefer these types of selectors over nth selectors if order isn't important.

Reply:
They were working fine until I ran these with WooCommerce Bookings installed on the site. It add a 'Booking' tab to the customer account screen which breaks 'nth-child'.

Since we will be using these tests with various plugins, it may be good to not use position dependent selectors like 'nth-child'

Reply:
Totally makes sense.  Way better to have it be flexible!